### ISSUE TITLE
Page repeatedly reloads while I am reading it

### ISSUE THREAD

**Issue by Elliott599** [mozilla-mobile/firefox-ios#18833]
## Steps to reproduce - load this page, or for that matter any review on rogerebert.com
https://www.rogerebert.com/reviews/spaceman-movie-review-netflix-2024

### Device & build information
* Device: ? iPhone 13 mini, IOS 17.3 and prior version of IOS 17
* Build version: ? 122.2 (38550)
* First seen version: ? at least 2 updates prior to 122.2
### Notes
Attachments:

**Comment by darkCavalier11** [comment]
I tried it on my simulator
iPhone 15
iOS 17.2
branch: main

Not able to reproduce

https://github.com/mozilla-mobile/firefox-ios/assets/57953014/0936e17d-c2c2-4a4e-ba87-eabae8e15eb4

**Comment by Elliott599** [comment]
I just updated to FFox 123.1 and it still happens and I tested it with Spaceman as did you.  Please try this.  In the video of what you did you are flipping around and going to the bottom of the page and back to the top.  That seems to change how rapidly the unwanted refresh occurs.  Please just pull up the page scroll down two paragraphs and just let it sit there for about a minute.  It does the refresh on mine within about 40 s to 1 min.  Thank you.

**Comment by LanceUEP** [comment]
I can confirm this bug. Every night I load firefox and go to google news. I “open in a new tab” all of the articles I want to read. Then I go through the tabs one by one reading them. 

Firefox has always refreshed multiple times while reading pages and I’ve always thought it was a stupid tracker or site related issue meant to stop ad blockers or something. 

I finally tonight decided to google the issue and ended up here.

**Comment by anatsymbol** [comment]
This is a constant issue for me as well. It seems to have something to do with ads or how Firefox handles pages that continue to load dynamically as you scroll down. This is not an issue in mobile Safari or any other 3rd party browser I've used on iOS.

**Comment by drider4** [comment]
I have had the same problem for months on tomshardware.com as well as some other sites. Sometimes rather than jumping all the way back to the beginning of an article it will go back to a position I had previously scrolled to.  Like it is only caching the distance in the article sometimes before the refresh occurs. 

I am now using Firefox 130 on an iPhone 13 with iOS 17.6.1

**Comment by Elliott599** [comment]
I have found a solution/work around on pages that support Reader View.  When I use Reader View the page reads fine and the reloading goes away.

**Comment by drider4** [comment]
> I have found a solution/work around on pages that support Reader View. When I use Reader View the page reads fine and the reloading goes away.

Yeah that’s what I do as well.  But I usually don’t think about it until the first time it jumps.  :)

**Comment by fedemorandi** [comment]
i solved this one (for me at least): literally close all the tabs and thats it

 I had apparently thousands of “opened tabs” -infinity icon showing in the tab indicator in the main firefox bar at the bottom. I used the trash icon to close all tabs.

I did it 3 times and firefox would close, and when I reopened it a lot of the tabs -and the infinity icon- were still there. Did it once more and the tabs actually closed -as opposed to the whole app-, saw a notification informing me that the tabs were closed and that solved it.

ive never closed (or actually opened) tabs before so who knows hoy many thousands of tabs where 
being stored there

hope this helps

**Comment by CDevine1** [comment]
For me it has nothing to do with how many tabs are opened. I frequently keep my tabs to less than 5 and always have the issue. When it happens I just go to safari as sometimes the reader mode doesn’t work very well.

**Comment by cwhillock** [comment]
Was having this issue on windowscentral.com and turning off night mode seemed to fix it

**Comment by mattolenik** [comment]
Still happening to me. iOS 18.5, iPhone 13 Mini, Firefox 140.1 (57242)

This didn't used to happen in past years.

**Comment by weisarom** [comment]
Concur that this is still happening. I notice it most often when reading https://www.askamanager.org/ posts. Page reloads again and again. This reloading often interferes with Find in Page. I enter the search string and the page reloads before doing the search, and the Find in Page dialog box disappears. 

Trying all this via Chrome on the same device works fine. 

ipados 18.5
9th Gen iPad
Firefox 141.1 (58972)

**Comment by Ostefanini** [comment]
The bug is present since a while now,
and it’s very inconvenient for people who stills read article on web pages

I’ve found a particular way to reproduce on this web page : https://www.presse-citron.net/essai-polestar-4-avis/

1 - zoom on the text, at maximum level
2 - scroll up and down

I’m pretty sure it’s related to advertising, because it never occurs on youtube.

The bug occurs even when not zooming, but it seems to be more reproductible this way

Iphone 13
iOS 18.6.2
Version 143.0.1

**Comment by sasimpson** [comment]
Same issue. Some pages will just reload constantly. Reader mode helps for articles. Does not do it with every page, seems to be ad-heavy pages that cause issues. 

iPhone 14Pro
iOS 26.0.1
Firefox 143.1.1 (62160)

https://github.com/user-attachments/assets/1b09db1a-9e22-4cb2-9a3f-d74694daabb3

**Comment by mattinsson** [comment]
Still a problem in firefox android 16. From google search results. Reloads after clicked and it is opened. Refreshes and its annoying.

**Comment by coderMohammed1** [comment]
I have the same issue on windows

**Comment by weisarom** [comment]
Is this ever going to be fixed??? Initial report 2 *years* ago and I chimed in Aug 2025. Accessing these same pages via Chrome  on the same device works problem free.

**Comment by PrizmaticAtmos** [comment]
Coming here to help folks understand. iOS requires that browsers use webkit. So chrome, and Firefox are using Safaris render engine. Not their own. The issue you are seeing should happen regardless of browser. My only explanation for why it may not occur in another browser is that a specific ad or dynamic content did not load that caused the crash in Firefox.

So essentially, this is not a Firefox issue but a safari webkit issue.

What is interesting is apple is allowing eu to use a different browser engine starting in 18.2

https://developer.apple.com/support/alternative-browser-engines/

**Comment by anatsymbol** [comment]
The posts specifically say this isn’t a problem on other iOS browsers. That’s my experience as well.

**Comment by PrizmaticAtmos** [comment]
I've had the same experience across multiple ios browsers on the same page when testing: Firefox, Chrome, vivaldi, and safari. I think your mileage may vary. Just sharing a possible explanation, I speak neither for Mozilla or Apple. Just another person suffering from this issue.
